### PR TITLE
Fix run_init for TPU command runner

### PR DESCRIPTION
**Pull request by allenwang28**
## Why are these changes needed?

https://github.com/ray-project/ray/commit/7060cfecf06b62432db1b404817b72d773c6f2bb introduced a change in the TPU command runner for TPU pods. This overrides the `run` command which previously returned a string, and now doesn't. Now if we try to use Docker within TPU runs, it will fail with an error message that `NoneType` does not have a function `decode()`. This fixes the string return which fixes the Docker issue.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

(and manual test)

### CODE REVIEW

File: python/ray/tests/gcp/test_gcp_tpu_command_runner.py [hunk 1]
```diff
@@ -188,6 +187,36 @@ def test_tpu_docker_command_runner():
...
+        "process_runner": process_runner,
+        "use_internal_ip": False,
+        "docker_config": docker_config,
+    }
+    cmd_runner = TPUCommandRunner(**args)
+
+    # Test for run_init
+    process_runner.respond_to_call("json .Config.Env", 2 * ["[]"])
+    cmd_runner.run_init(as_head=True, file_mounts={}, sync_run_yet=True)
+    process_runner.assert_has_call("1.2.3.4", pattern="docker")
```
Comment (architkulkarni) [inline python/ray/tests/gcp/test_gcp_tpu_command_runner.py:217]: This part is hard to understand and could use some comments.

**Review by architkulkarni — APPROVED** [review]
Looks good to me, let me know when tests are passing!

File: python/ray/tests/gcp/test_gcp_tpu_command_runner.py (hunk 1 above)
Comment (allenwang28) [reply python/ray/tests/gcp/test_gcp_tpu_command_runner.py:220]: added a comment, thanks!

Comment (architkulkarni) [reply python/ray/tests/gcp/test_gcp_tpu_command_runner.py:220]: Thanks, looks good!

**Pull request merged**